### PR TITLE
versions: Update kernel to version v6.1.x

### PR DESCRIPTION
**Pull request by jepio**
Kernel v6.1.38 is the current latest LTS version, switch to it.  No patches should be necessary. CONFIG_MEMCG_SWAP has been removed.

Fixes: #6086

### CODE REVIEW

**Review by fengwang666 — APPROVED** [review]
lgtm. Thanks!

**Comment by fidencio** [comment]
@jepio, mind to rebase your PR atop of `main` in order to solve conflicts?

**Comment by zvonkok** [comment]
I would need to test if this works with NV GPU + SNP and the regular use-case where I do not see any issues. I am curious if this was tested with the non-UPM host kernel, since 6.xx is adding UPM support.

**Comment by zvonkok** [comment]
@ryansavino Can a host non-UPM kernel work with a UPM guest kernel?

**Comment by jepio** [comment]
> I would need to test if this works with NV GPU + SNP and the regular use-case where I do not see any issues. I am curious if this was tested with the non-UPM host kernel, since 6.xx is adding UPM support.

There is no dependency between SNP host and SNP guest kernel, I tested 5.19-6.3 guest kernels with non-UPM hosts.

> @ryansavino Can a host non-UPM kernel 5.19.xxx work with a UPM guest kernel 6.1.xxx ? Are the UPM fixes already in mainline "v6.1.27" ?

UPM is not merged upstream, so i don't know which UPM fixes you might be referring to.

**Comment by zvonkok** [comment]
> UPM is not merged upstream, 
This was one of my questions. 

Did you test 6.1.27 guest kernel on non-UPM host?

**Comment by zvonkok** [comment]
Give me a day or two to test it and then we're good to go.

**Comment by ryansavino** [comment]
> @ryansavino Can a host non-UPM kernel 5.19.xxx work with a UPM guest kernel 6.1.xxx ? Are the UPM fixes already in mainline "v6.1.27" ?

UPM host requires UPM guest. I haven't tried UPM guest on non-UPM host. I can gather more details on this today, but it sounds like @jepio has it working?

**Comment by jepio** [comment]
@fidencio since the release happened, lets revisit this.
@zvonkok have you tested this?

Since we used a mainline v5.19 before, upgrading to mainline v6.1 should not introduce any regressions.

**Comment by zvonkok** [comment]
```
+ suffix=-nvidia-gpu-sev
+ vmlinuz=vmlinuz-6.1.38-111-nvidia-gpu-sev
+ vmlinux=vmlinux-6.1.38-111-nvidia-gpu-sev
+ '[' -e arch/x86_64/boot/bzImage ']'
```

Everything is build correct ...

**Comment by jepio** [comment]
> After building the GPU kernels I end up with
> 
> ```
> drwxr-xr-x root/root         0 2023-07-14 10:59 ./opt/kata/share/kata-containers/
> lrwxrwxrwx root/root         0 2023-07-14 10:59 ./opt/kata/share/kata-containers/vmlinuz-nvidia-gpu-sev.container -> vmlinuz-5.19.2-109-nvidia-gpu-sev
> -rw-r--r-- root/root   6879488 2023-07-14 10:59 ./opt/kata/share/kata-containers/vmlinuz-5.19.2-109-nvidia-gpu-sev
> -rw-r--r-- root/root  56267264 2023-07-14 10:59 ./opt/kata/share/kata-containers/vmlinux-5.19.2-109-nvidia-gpu-sev
> -rw-r--r-- root/root     79331 2023-07-14 10:59 ./opt/kata/share/kata-containers/config-5.19.2-109-nvidia-gpu-sev
> lrwxrwxrwx root/root         0 2023-07-14 10:59 ./opt/kata/share/kata-containers/vmlinux-nvidia-gpu-sev.container -> vmlinux-5.19.2-109-nvidia-gpu-sev
> ```
> 
> The builds are indeed with the 6.x kernel but something is not picking up the latest version. Checking.

I don't see this in the CI, could it be a local caching issue?

**Comment by zvonkok** [comment]
@jepio Yes I did not delete the `kata-static-kernel-...tar.gz` Now looking good. I can build the kernel. 
Need to build a new intird and test CC GPU.

**Comment by zvonkok** [comment]
Ok at least for the CC GPU this kernel fails to boot. Need to investigate why this happens. The other possibility would be to merge this and update the GPU kernel functions to use 5.19.2 until it is fixed.

File: versions.yaml [hunk 1]
```diff
@@ -168,16 +168,16 @@ assets:
...
-    url: "https://cdn.kernel.org/pub/linux/kernel/v5.x/"
-    version: "v5.19.2"
+    url: "https://cdn.kernel.org/pub/linux/kernel/v6.x/"
+    version: "v6.1.38"
     sev:
       description: "Linux kernel that supports SEV and SNP"
-      url: "https://cdn.kernel.org/pub/linux/kernel/v5.x/"
-      version: "v5.19.2"
+      url: "https://cdn.kernel.org/pub/linux/kernel/v6.x/"
+      version: "v6.1.38"
```
Comment (fidencio) [inline versions.yaml:176]: Based on @zvonkok comment, let's not update this one yet.

**Review by fidencio — COMMENTED** [review]
@jepio, please, do not update SEV / SNP versions for now as it breaks GPUs on that.

File: versions.yaml (hunk 1 above)
Comment (jepio) [reply versions.yaml]: pulled this back down to v5.19.2

File: versions.yaml [hunk 2]
```diff
@@ -168,16 +168,16 @@ assets:
...
     snp:
       description: "Linux kernel that supports AMD SEV-SNP for VMs"
-      url: "https://cdn.kernel.org/pub/linux/kernel/v5.x/"
-      version: "v5.19.2"
+      url: "https://cdn.kernel.org/pub/linux/kernel/v6.x/"
+      version: "v6.1.38"
```
[the remark this answers is not in the document]
Comment (jepio) [reply versions.yaml]: pulled this back down to v5.19.2

**Pull request merged**